### ISSUE TITLE
direct-volume: file name too long

### ISSUE THREAD

**Issue by ChengyuZhu6** [kata-containers/kata-containers#10284]
# Description of problem

 the generated file path from kata direct volume exceeds the maximum length allowed by the file system. This is caused by using `base64.URLEncoding` to encode the `volumePath`, resulting in a very long directory and file name.

error log: 
```
failed to parse the mount info file for a direct assigned volume" container=26f2b0953b3e17241679a78ac7e8b55e2f9ad558e6475d7957db08d1846154c1 error="stat /run/kata-containers/shared/direct-volumes/L3Zhci9saWIvY29udGFpbmVyZC9pby5jb250YWluZXJkLmdycGMudjEuY3JpL2NvbnRhaW5lcnMvMjZmMmIwOTUzYjNlMTcyNDE2NzlhNzhhYzdlOGI1NWUyZjlhZDU1OGU2NDc1ZDc5NTdkYjA4ZDE4NDYxNTRjMS92b2x1bWVzLzU3OWY3Y2RkOWI4NmQ0ZmM2OWIzOTlmZGM5MTY4ODViODMwZWNjOTgzNzc4NDI4Y2IwYzc3MWEzZTUxMmNiZTk=/mountInfo.json: file name too long" mount-source=/var/lib/containerd/io.containerd.grpc.v1.cri/containers/26f2b0953b3e17241679a78ac7e8b55e2f9ad558e6475d7957db08d1846154c1/volumes/579f7cdd9b86d4fc69b399fdc916885b830ecc983778428cb0c771a3e512cbe9 name=containerd-shim-v2 pid=804276 sandbox=d861add7b933adc4464b83886c90c1175cbe2c2008bdb79c8ae52a1a16280e15 source=virtcontainers subsystem=container
```

**Comment by Apokleos** [comment]
Hi @ChengyuZhu6 Thx for reporting this issue. I think we need know more about it.
What's your develop ENV, or especially what's the File system you're using to run kata ?

**Comment by ChengyuZhu6** [comment]
> Hi @ChengyuZhu6 Thx for reporting this issue. I think we need know more about it. What's your develop ENV, or especially what's the File system you're using to run kata ?

Hi @Apokleos That's my system info:

Operating System: Ubuntu 22.04.4 LTS
 OSType: linux
 Architecture: x86_64

**Comment by ChengyuZhu6** [comment]
> > > Hi @ChengyuZhu6 Thx for reporting this issue. I think we need know more about it. What's your develop ENV, or especially what's the File system you're using to run kata ?
> > 
> > 
> > Hi @Apokleos That's my system info:
> > Operating System: Ubuntu 22.04.4 LTS OSType: linux Architecture: x86_64
> 
> Thx @ChengyuZhu6 I have the almost same DevEnv(Ubuntu 22.04.3 LTS), could you please provide more info about how to reproduce it ?
> 
> I need to confirm the problem, as it involves several components. Thx again.

I fully follow the document to use direct-volume: https://github.com/kata-containers/kata-containers/blob/main/src/tools/csi-kata-directvolume/docs/deploy-csi-kata-directvol.md

**Comment by Apokleos** [comment]
> > > > Hi @ChengyuZhu6 Thx for reporting this issue. I think we need know more about it. What's your develop ENV, or especially what's the File system you're using to run kata ?
> > > 
> > > 
> > > Hi @Apokleos That's my system info:
> > > Operating System: Ubuntu 22.04.4 LTS OSType: linux Architecture: x86_64
> > 
> > 
> > Thx @ChengyuZhu6 I have the almost same DevEnv(Ubuntu 22.04.3 LTS), could you please provide more info about how to reproduce it ?
> > I need to confirm the problem, as it involves several components. Thx again.
> 
> I fully follow the document to use direct-volume: https://github.com/kata-containers/kata-containers/blob/main/src/tools/csi-kata-directvolume/docs/deploy-csi-kata-directvol.md
>
Yeah, It's written by me. But it's so puzzling, I have ever tested it many times without such error.

**Comment by sprt** [comment]
@ChengyuZhu6 Can you provide a repro? I've also never experienced such an issue with block devices.

The base64 string in your initial message decodes to the below:

```
/var/lib/containerd/io.containerd.grpc.v1.cri/containers/26f2b0953b3e17241679a78ac7e8b55e2f9ad558e6475d7957db08d1846154c1/volumes/579f7cdd9b86d4fc69b399fdc916885b830ecc983778428cb0c771a3e512cbe9
```

This seems surprising to me as usually that string should only be a basename?
